### DISCUSSION TITLE
Headless data handling...

### DISCUSSION THREAD

**Discussion by rchrdnsh** [sveltia/sveltia-cms#48] [Ideas]
So, really liking Sveltia CMS for general usage, but I would love to be able to make custom form pages for complex data requirements, so as to create a more specific and streamlined user experience for those entering that data.

Here is a custom form UI I am making for a specific data upload usecase:

<img width="1472" alt="Screenshot 2023-08-26 at 11 56 46 PM" src="https://github.com/sveltia/sveltia-cms/assets/7884263/a7f6eb3a-c611-4245-a0df-2fa2e1590263">

While Sveltia is great for general purpose data, something like this, without explaining it too much, is a bit cumbersome using a standard one column data entry ui layout style.

So I'm thinking it would be really interesting if there was some sort of...something...a library or shell of functionality maybe...where the whole uploading and saving files that have been edited as...i don't know how this cms works, to be honest...but it might be cool if I were to be able to build custom ui's like this on top of the core file handling and saving capabilitites of sveltia...

Don't know if that makes any sense or not...just something that has been on my mind :-)

**Comment by kyoshino** [comment]
I think this kind of thing is possible in Netlify/Decap CMS using [custom widgets](https://decapcms.org/docs/custom-widgets/). Sveltia CMS has not added the support yet, but it’s on the roadmap for sure ☺️
